system-contracts/test/NonceHolder.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { expect } from "chai";
+import type { NonceHolder } from "../typechain";
+import { NonceHolderFactory } from "../typechain";
+import {
+  TEST_BOOTLOADER_FORMAL_ADDRESS,
+  TEST_DEPLOYER_SYSTEM_CONTRACT_ADDRESS,
+  TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS,
+} from "./shared/constants";
+import { prepareEnvironment, setResult } from "./shared/mocks";
+import { deployContractOnAddress, getWallets } from "./shared/utils";
+import { ethers, network } from "hardhat";
+
+describe("NonceHolder tests", () => {
+  const wallet = getWallets()[0];
+  let nonceHolder: NonceHolder;
+  let nonceHolderAccount: ethers.Signer;
+  let deployerAccount: ethers.Signer;
+
+  before(async () => {
+    await prepareEnvironment();
+    await deployContractOnAddress(TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS, "NonceHolder");
+    nonceHolder = NonceHolderFactory.connect(TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS, wallet);
+    nonceHolderAccount = await ethers.getImpersonatedSigner(TEST_BOOTLOADER_FORMAL_ADDRESS);
+    deployerAccount = await ethers.getImpersonatedSigner(TEST_DEPLOYER_SYSTEM_CONTRACT_ADDRESS);
+  });
+
+  after(async () => {
+    await network.provider.request({
+      method: "hardhat_stopImpersonatingAccount",
+      params: [TEST_DEPLOYER_SYSTEM_CONTRACT_ADDRESS],
+    });
+
+    await network.provider.request({
+      method: "hardhat_stopImpersonatingAccount",
+      params: [TEST_BOOTLOADER_FORMAL_ADDRESS],
+    });
+  });
+
+  describe("increaseMinNonce", () => {
+    it("should revert This method require system call flag", async () => {
+      await expect(nonceHolder.increaseMinNonce(123)).to.be.rejectedWith("This method require system call flag");
+    });
+
+    it("should increase account minNonce", async () => {
```

Comment:
Looks like this test case can not be run twice without a node reboot. Can you check the nonce change instead certain value or use a random wallet?

Reply:
done